prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -2227,8 +2229,14 @@ ssize_t rxr_pkt_init_rta(struct rxr_ep *ep, struct rxr_op_entry *tx_entry,
 
 	hdr_size = rxr_pkt_req_hdr_size_from_pkt_entry(pkt_entry);
 	data = (char *)pkt_entry->pkt + hdr_size;
-	data_size = ofi_copy_from_iov(data, ep->mtu_size - hdr_size,
-				      tx_entry->iov, tx_entry->iov_count, 0);
+	hmem_iface = ((struct efa_mr*) tx_entry->desc[0])->peer.iface;
```

Comment:
`tx_entry->desc[0]` can be NULL for system memory

Reply:
It will never be NULL.  Atomics are for one sided communication, and in order to use 1 sided communication, you must register all your memory.

Reply:
The compare, result, and buff must all be registered no-matter the memory type in order to use libfabric atomics AFAIK... I think an assert would be better here if you are concerned.